openarena-q3e

```diff
@@ -1,29 +1,66 @@
 #!/bin/sh
-OA_CVARS="+set fs_basegame baseoa +set protocol 71"
-DISABLED_CVARS="+set com_protocol 72 +set cl_voip 1"
 
+###
+CONFIG_FILE=".var/app/io.github.ec_.Quake3e.OpenArena/config/cfg"
+# Check if the config file exists
+if [ -f "$CONFIG_FILE" ]; then
+    source "$CONFIG_FILE"
+fi
+# Check if it's the first run or if RESPECT_XDGDIRS is not set
+if [ ! -f "$CONFIG_FILE" ] || [ -z "$RESPECT_XDGDIRS" ]; then
+    RESPECT_XDGDIRS=0
+
+    # Save RESPECT_XDGDIRS in the config file
+    echo "RESPECT_XDGDIRS=$RESPECT_XDGDIRS" > "$CONFIG_FILE"
+    echo -e "It seems this is the first time you run OpenArena... You can edit the $$CONFIG_FILE and change RESPECT_XDGDIRS to use your XDG_DATA_FOLDER, which is: ${XDG_DATA_HOME}/homepath Instead of polluting your HOMEDIR by adding the hidden .openarena folder. \n"
+fi
+# Use the value of RESPECT_XDGDIRS to determine the directory
+if [ -n "$RESPECT_XDGDIRS" ] && [ "$RESPECT_XDGDIRS" -eq 1 ]; then
+    # Use ${XDG_DATA_HOME}/homepath directory
+    mkdir -p "${XDG_DATA_HOME}/homepath"
+    OA_DIR="${XDG_DATA_HOME}/homepath"
+    if [ -d "$HOME/.openarena" ]; then
+        mv "$HOME/.openarena"/* "$OA_DIR"
+        rm -rf "$HOME/.openarena"
+    fi
+    if [ -d "$HOME/.var/app/io.github.ec_.Quake3e.OpenArena/.openarena" ]; then
+        rm -rf "$HOME/.var/app/io.github.ec_.Quake3e.OpenArena/.openarena"
+    fi
+else
+    # Use .openarena directory
+    OA_DIR="$HOME/.openarena"
+    rm -rf "${XDG_DATA_HOME}/homepath"
+fi
+###
+
+OA_CVARS="+set fs_basegame baseoa +set protocol 71 +set fs_homepath $OA_DIR"
+DISABLED_CVARS="+set com_protocol 72 +set cl_voip 1"
 arch="$(uname -m)"
 case "$arch" in
     x86_64) q3e_arch="x64" ;;
     aarch64) q3e_arch="aarch64" ;;
 esac
 
+if [ "$RESPECT_CONVENTIONS" != "" ] && [ "$RESPECT_CONVENTIONS" -eq 0 ]; then
+    echo -e "If you'd rather change your homepath from ~/.openarena to ~/.var/app/io.github.ec_.Quake3e.OpenArena/homepath you can set the variable RESPECT_CONVENTIONS to 1 in the following config file: ~/.var/app/io.github.ec_.Quake3e.OpenArena/config/cfg \n"
```

Comment:
Also incorrect and misleading for users. Keep in mind that while this path exists inside the Flatpak, it is not the actual path for end-users to find the files on their filesystem (since they are sandboxed). Also, you have typo here (the actual path is `data/homepath`, not just `homepath`).

The actual echo message should probably be:
`echo -e "If you'd rather change your homepath from ~/.var/app/io.github.ec_.Quake3e.OpenArena/.openarena to ~/.var/app/io.github.ec_.Quake3e.OpenArena/data/homepath you can set the variable RESPECT_CONVENTIONS to 1 in the following config file: ~/.var/app/io.github.ec_.Quake3e.OpenArena/config/cfg \n"`

Reply:
Not really, the persist folder is in ~/.openarena

Reply:
Yes I forgot to change homepath to data/homepath/